src/Idris/Parser.hs

```diff
@@ -1124,8 +1103,8 @@ WhereOrTerminator ::= WhereBlock | Terminator;
 -}
 clause :: SyntaxInfo -> IdrisParser PClause
 clause syn
-         = do wargs <- P.try (do pushIndent; some (wExpr syn))
-              fc <- getFC
+         = appExtent (do
```

Comment:
What does `appExtent` do?

Reply:
Well, the neat thing about `withExtent` is that we piggieback on the composition of parsers to compose how to compute the FC that spans the parse.  The problem is that there are a few cases where we need the FC deep inside some value composed inside the parsers; so `appExtent` expects a value abstracted over the FC and applies the FC to it.  It's named after the `app` combinator (but it definitely isn't the greatest name).